src/logs/sentry.rs

```diff
@@ -94,119 +104,183 @@ impl IntoIterator for LazySentryLogs {
     }
 }
 
+pub trait Stat {
+    fn check_raw_log(&mut self, log: &RawLog);
+    fn report(&self);
+    fn process_raw_logs<'l>(&mut self, logs: impl Iterator<Item = &'l RawLog>) {
+        for log in logs {
+            self.check_raw_log(log);
+        }
+    }
+}
+
+pub struct SuccessfulScan {
+    pub total: usize,
+}
+
+pub struct MalformedQr {
+    pub total: usize,
+}
+
+pub struct RegexMatch {
+    key: String,
+    re: Regex,
+    total_checked: usize,
+    pub matches: usize,
+}
+
+pub enum SentryLogsStatChecker {
```

Comment:
I think trait objects could be used without the need to have an enum, but that's just personal preference

Reply:
Yeah, but the idea behind this was to be able to access some public internal attributes if needed. Which you can't do with traits directly iirc.